ownership/sync_repos.py

```diff
@@ -0,0 +1,286 @@
+        elif found_repo.archived:
+            recommend_delete_row(ss_name, "Repo has been archived")
+        else:
+            new_name = str(found_repo).lower()
+            if [r for r in repos_ss if r['name'] == new_name]:
+                raise KnownError(
+                    "Manual intervention required: A repo that has moved "
+                    "exists in the spreadsheet under both its old name "
+                    "and its new name. %s -> %s"
+                     % (ss_name, new_name))
+            actions.append({
+                'action': 'rename_row',
+                'repo': ss_name,
+                'new_name': new_name,
+                'row_id': find_row_by_name(ss_name)['row_id'],
+                'why': ("Repo has been renamed, and is not already in the "
+                        "spreadsheet under that name")
+            })
+            names_only_ss.remove(ss_name)
+            names_only_gh.remove(new_name)
+
+    # Now check ones that are only in Github. We've already filtered
+    # out archived repos and removed any that are pending a rename on
+    # the spreadsheet.
+    for gh_name in names_only_gh.copy():
+        actions.append({
+            'action': 'add_row',
+            'repo': gh_name,
+            'why': "Repo present in Github but not spreadsheet"
+        })
+        names_only_gh.remove(gh_name)
+
+    # Make sure nothing is unaccounted for before presenting recommendations.
+    if names_only_gh or names_only_ss:
+        raise KnownError(
+            "Failed to convert all mismatches between Github and spreadsheet "
+            "into actions! Github remaining = %s; spreadsheet remaining = %s"
+            % (names_only_gh, names_only_ss))
+
+    # Ensure actions don't have any internal conflicts. It might be OK
+    # if the same *name* comes up twice; for example, we might want to
+    # move an archived repo to the unspported org *and* remove it from
+    # the spreadsheet. However, two actions for the same row ID is a
+    # problem.
+    seen_row_ids = set()
+    for action in actions:
+        row_id = action.get('row_id')
+        if not row_id:
+            continue
+        if row_id in seen_row_ids:
+            raise KnownError(
+                "Possible bug in script: Was going to suggest multiple "
+                "actions for row %s: %s"
+                % (row_id, [r for r in actions if r.get('row_id') == row_id]))
+        seen_row_ids.add(row_id)
+
+    return actions
+
+
+def report_actions(actions):
+    """Report the recommended actions without making changes."""
+    if actions:
+        print("Recommendations:")
+
+        groupkey = lambda a: (a['action'], a['why'])
+        for ((action_name, why), group) in groupby(sorted(actions, key=groupkey), groupkey):
+            print()
+            print("Action: %s -- %s" % (action_name, why))
+            for action in group:
+                data = action.copy()
+                del data['action']
```

Comment:
I think a comment here on why you're doing these deletes would be useful.  I think you can just take the explanation from your github comment https://github.com/edx/jenkins-job-dsl/pull/1004#discussion_r437057700

FWIW, the onliner was pretty easy to read for me so I would be fine with that as well along with the explanation of why.